c/tskit/trees.c

```diff
@@ -2012,6 +2018,113 @@ tsk_treeseq_trait_correlation(tsk_treeseq_t *self,
     return ret;
 }
 
+static int
+trait_regression_summary_func(size_t state_dim, double *state, size_t result_dim,
+        double *result, void *params)
+{
+    covariates_stat_params_t args = *(covariates_stat_params_t *) params;
+    const double num_samples = (double) args.num_samples;
+    const tsk_size_t k = args.num_covariates;
+    const double *V = args.V;;
+    const double *x = state;
+    const double *v;
+    double m, a, denom, xx;
+    size_t i, j;
+
+    m = x[state_dim - 1];
+    for (i = 0; i < result_dim; i++) {
+        if ((m > 0.0) && (m < num_samples)) {
+            v = GET_2D_ROW(V, k, i);
+            a = x[i];
+            denom = m;
+            for (j = 0; j < k; j++) {
+                xx = x[result_dim + j];
```

Comment:
This should be ``state_dim``, shouldn't it?

Reply:
Nope - here, `x[0], ..., x[result_dim - 1]` is the internal state corresponding to the traits, `W`, and `x[result_dim], ..., x[state_dim - 2]` is the state corresponding to the covariates, `Z`, and `x[state_dim - 1]` is the "tracked sample count" column. (But, maybe I should have called this `z` instead of `xx`.)

Reply:
I see, thanks for clarifying